app/models/katello/glue/pulp/repo.rb

```diff
@@ -255,7 +255,7 @@ def generate_distributors(capsule = nil)
           dist.auto_publish = true
           distributors = [dist]
         when Repository::PUPPET_TYPE
-          capsule ||= SmartProxy.default_capsule
+          capsule ||= SmartProxy.default_capsule!
```

Comment:
The trick is to look at the different between these two methods. This version throws a nice error message the user can use to debug, without it the user simply gets `no method puppet_path for nil class`